src/secure_random.cr

```diff
@@ -1,19 +1,60 @@
 require "base64"
 require "openssl/lib_crypto"
 
+# The SecureRandom library is an interface for creating secure random values in different formats, 
+# including Base64 string, hexadecimal string, and uuid, etc.
+#
+# For example:
+# ```crystal
+# SecureRandom.base64 #=> LIa9s/zWzJx49m/9zDX+VQ==
+# SecureRandom.hex    #=> c8353864ff9764a39ef74983ec0d4a38
+# SecureRandom.uuid   #=> c7ee4add-207f-411a-97b7-0d22788566d6
+# ```
 module SecureRandom
+  # Generates a random base64 string.
+  #
+  # The argument `n` determines its size, which is around `n` * 1.33.
+  # The returned string may contain A-Z, a-z, 0-9, +, /, and =
+  # 
+  # ```crystal
+  # SecureRandom.base64(4) #=> fK1eYg==
+  # ```
   def self.base64(n = 16)
     Base64.strict_encode(random_bytes(n))
   end
 
+  # Same as `#base64` but is URL-safe.
+  #
+  # In addition, an extra argument `padding`(boolean) decides if padding `==` is generated. (Default: false)
+  # 
+  # ```crystal
+  # SecureRandom.urlsafe_base64 #=> MAD2bw8QaBdvITCveBNCrw
+  # SecureRandom.urlsafe_base64(16,true) #=> og2aJrELDZWSdJfVGkxNKw==
+  # ```
   def self.urlsafe_base64(n = 16, padding = false)
     Base64.urlsafe_encode(random_bytes(n), padding)
   end
 
+  # Generates a random hexadecimal string.
+  # 
+  # The argument `n` determines its size, which is `n` * 2.
+  # The returned string may contain a-z, 0-9.
```

Comment:
This is a hexstring, which means that each byte with be under its hexadecimal form, going from `00` to `ff` so chars are `[0-9a-f]`. Hence why the total string size is twice the number of bytes.